metadata/metadata.go

```diff
@@ -1865,6 +1870,34 @@ func (serv *MetadataServer) genericCreate(ctx context.Context, res Resource, ini
 	return &pb.Empty{}, nil
 }
 
+func (serv *MetadataServer) setDefaultVariant(id ResourceID, defaultVariant string) error {
+	parent, err := serv.lookup.Lookup(id)
+	if err != nil {
+		return err
+	}
+	var parentResource Resource
+	if resource, ok := parent.(*SourceResource); ok {
+		resource.serialized.DefaultVariant = defaultVariant
+		parentResource = resource
+	}
+	if resource, ok := parent.(*labelResource); ok {
```

Comment:
should these not be else/if or switch?

Reply:
If i do the switch I cant access the fields I want